### ISSUE TITLE
Conformance tests do not generate any transactions

### ISSUE THREAD

**Issue by dnadales** [input-output-hk/cardano-ledger-byron#698]
## Background

In the executable specification we use a high value for the `b` constant used in the calculation of the minimum transaction fee. The reason for this is explained [here](https://github.com/input-output-hk/cardano-ledger-specs/blob/2aa807f4a4b6fd0ae0a511289586a2c2d18b56fb/byron/ledger/executable-spec/src/Ledger/Update.hs#L859). The problem with this approach, is that the [UTxO generators](https://github.com/input-output-hk/cardano-ledger-specs/blob/09ca95901d6b1690f10886e690d1adb74df40923/byron/ledger/executable-spec/src/Ledger/UTxO/Generators.hs#L31) generate a balance [between 1 and 10000](https://github.com/input-output-hk/cardano-ledger-specs/blob/09ca95901d6b1690f10886e690d1adb74df40923/byron/ledger/executable-spec/src/Ledger/UTxO/Generators.hs#L39). As a consequence, the UTxO generators can't possible produce a transaction that can cover the minimum fee, which means that no UTxO is generated in the `CHAIN` traces.

We have coverage tests of UTxO traces, however these traces use a different generator for initial parameters, which sets the `b` value between 0 and 10 (as opposed to the constant 10 * 4096 used in the `CHAIN` traces).

## Proposed Solution

A proper solution would be to consider the protocol parameters `a` and `b` (or the minimum fee), in the function that generates the initial UTxO, so that it generates a balance high enough to cover the minimum fee. 

We should also add coverage tests for `CHAIN` traces, to ensure that we're generating blocks with  UTxO payload.

**Comment by edsko** [comment]
Specifically, it means that the conformance testing between the spec and implementation do _not_ currently use blocks that contain any regular transactions at all. 

Currently trying to work around the problem in the consensus tests for the Byron ledger (https://github.com/input-output-hk/ouroboros-network/pull/1503), which is where I first discovered this problem in the first place. If the workaround works, at least that means we are testing conformance there.

**Comment by edsko** [comment]
Related problem: the value of `c `in `Ledger.GlobalParams` is not compatible with the value that the UTXOW rule assumes for `b`. The UTxOW assumes `0 <= b <= 10`, but the value of `c` is currently set to `4096`, which means that if we pick `0 <= b <= 10` for the spec, we'd end up with `bC` equal to `0` always (see `elaborateFeePolciy`).

**Comment by edsko** [comment]
Ok, I think I understand the problem better now. Let me try to give a cohesive description:

This bug arises from the interaction between two rules: UTXOW and CHAIN.

* The UTXOW rule calls `pparamsGen` to produce protocol parameters. This function picks  

  ```
  0 <= a <= 100
  0 <= b <= 10
  ```

  for the fee parameters. When it produces outputs for the initial UTxO, it chooses correspondingly small values, between `0` and `10,000`. This is all fine, by itself.

* When we relate the Byron specification to the Byron implementation, the translation of these fee parameters is problematic: the `b` parameter determines the fee of the transaction _in terms of the concrete size_ of that transaction, but of course the concrete size of an abstract transaction is basically unrelated to the concrete size of a real transaction.

* In order to account for this, the spec introduces a parameter `c` with property that

  ```
  size (elaborate a) <= c * abstractSize a
  ```

  If we then set the `b` parameter for the spec `c` times higher than for the implementation, the elaboration of a spec transaction should never result in a real transaction with insufficient fees (provided that the above property actually holds).

  The spec sets `c = 4096`.

* The `CHAIN` rule then picks initial protocol parameters for the spec that where `b` is set to `10 * c`, resulting in a `b` parameter for the real implementation of `10`. However, _it does not adjust the UTxO_ (generated in the UTXOW rule). As a consequence, when the UTXOW tries to produce transactions, there is a `c` factor mismatch between the UTxO values it is working with and the transaction fee parameters, and so every transaction ever generated is rejected (by the generator itself) as not covering the fees.

It seems to me that the solution here is simply to apply that `c` correction factor to the UTxO as well. I will try this in our consensus tests.

**Comment by edsko** [comment]
Fixed in #1177.
